### PR TITLE
Return header directly instead of wrapper struct

### LINKED ISSUE #487: ObjectHead must return object header
We return it in a simple way even for ObjectGet, going via ResObjectHead is counterproductive.

### PR DESCRIPTION
**Pull request by smallhive**
closes #487

### CODE REVIEW

File: client/object_get.go [hunk 1]
```diff
@@ -448,7 +448,12 @@ func (c *Client) ObjectHead(ctx context.Context, containerID cid.ID, objectID oi
 		res.hdr = v
 	}
 
-	return &res, nil
+	var hdr object.Object
+	if !res.ReadHeader(&hdr) {
+		return nil, errors.New("empty header")
```
Comment (roman-khimov) [inline client/object_get.go:453]: Can't there be some protocol-level error there? @carpawell?

Comment (carpawell) [reply client/object_get.go:453]: @roman-khimov, you mean that we can return some "bad response" error here? or you mean that header is missing cause we have not checked it?

1. there is no such err, but we can add smth like https://github.com/nspcc-dev/neofs-api/issues/269 but for client side
2. no, we checked statuses on `processResponse` call and no error retuned so we are looking for a header

File: client/object_get.go [hunk 2]
```diff
@@ -349,7 +349,7 @@ type ResObjectHead struct {
 
 // ReadHeader reads header of the requested object.
 // Returns false if header is missing in the response (not read).
-func (x *ResObjectHead) ReadHeader(dst *object.Object) bool {
+func (x *resObjectHead) ReadHeader(dst *object.Object) bool {
```
[the remark this answers is not in the document]
Comment (roman-khimov) [reply client/object_get.go:352]: `resObjectHead` is private, so it doesn't matter much.

File: client/object_get.go (hunk 1 above)
Comment (roman-khimov) [reply client/object_get.go:453]: Can there be some error returned from the server? Like we're trying to read from a stream, but server closes it for whatever reason.

Comment (carpawell) [reply client/object_get.go:453]: can't say i understand you. `ReadHeader` does not do any RPC, just struct conversions. all the network work has been done (and checked) above

File: client/object_get.go (hunk 2 above)
[the remark this answers is not in the document]
Comment (carpawell) [reply client/object_get.go:352]: sure but usually when i see public methods on a private struct i think that it implements something and all. that is just an internal converter. if there is no external tests for it, i suggest to make it private. do not insist

File: client/object_get.go (hunk 1 above)
Comment (roman-khimov) [reply client/object_get.go:453]: Ah, I see now. Then we don't need `resObjectHead` at all.

Comment (carpawell) [reply client/object_get.go:453]: > don't need resObjectHead at all

yeap, that is why i requested dropping its publicity. but complete dropping is even sweeter

**Comment by smallhive** [comment]
Removed resObjectHead completely

**Comment by carpawell** [comment]
So to draft or not to draft? That is the question.

**Comment by smallhive** [comment]
No draft, of course :sweat_smile:

**Review by roman-khimov — APPROVED** [review]

**Review by carpawell — APPROVED** [review]

**Comment by roman-khimov** [comment]
Conflicts after #491 merge.

**Comment by smallhive** [comment]
Rebased

**Pull request merged**